src/main/java/com/mparticle/kits/ButtonKit.java

```diff
@@ -73,165 +91,222 @@ public ButtonKit getInstance() {
         return null;
```

Comment:
In what scenario would a partner app call `setOptOut(true)`? Should we be calling `clearAllData()` if they do?